src/compiler.ts

```diff
@@ -8682,12 +8682,36 @@ export class Compiler extends DiagnosticEmitter {
     var values = expression.values;
     var members = classReference.members;
     var hasErrors = false;
-    var exprs = new Array<ExpressionRef>(numNames + 2);
+    var exprs = new Array<ExpressionRef>();
     var flow = this.currentFlow;
     var tempLocal = isManaged
       ? flow.getAutoreleaseLocal(classReference.type)
       : flow.getTempLocal(classReference.type);
     assert(numNames == values.length);
+
+    // Assume all class fields will be omitted, and add them to our omitted list
+    var omittedClassFieldMembers = new Map();
+    if (members) {
+      for (let memberKey of members.keys()) {
+        let member = members.get(memberKey);
+        if (member && member.kind == ElementKind.FIELD) {
+          omittedClassFieldMembers.set(member.name, '');
```

Comment:
Wondering if this can be a `Set`, if initial values are empty?

Reply:
Yeah I originally was thinking a set, but then I got distracted and used a map when I came back :upside_down_face: Will switch to a set! :smile:

Reply:
Oh wait! I remember, I chose not to use a set, because Set.values() returns am Iterator in JS, but an Array in AS :upside_down_face:

Reply:
Yeah, `Map`s and `Set`s have the same problem there. For `Set`s, the helper is `Set_values`. Both can't be iterated in a uniform way in TS/AS yet (no `for ... of`).

Reply:
Oh wait ignore me, Map.keys() also returns an iterator haha!